README.md

```diff
@@ -39,35 +51,131 @@ class ListMovingVehicle(vqpy.QueryBase):
 
 ```python
 vqpy.launch(cls_name=vqpy.COCO_CLASSES, # detection class
             cls_type={"car": Vehicle, "truck": Vehicle}, # mappings from detection class to VObj
             tasks=[ListMovingVehicle()], # a list of Queries to apply
             video_path=args.path, # the path of the queried video
-            save_folder=args.save_folder # result of query will be saved as a json file in this folder
+            save_folder=args.save_folder, # result of query will be saved as a json file in this folder
+            detector_name="yolox", # optional, specify the name of user's own model
+            detector_model_dir=args.pretrained_model_dir # optional, specify the directory of user's pretrained model
             )
 ```
 
-Under the hood, VQPy will automatically select an object detection model that outputs the specified `cls_name`. Multiple video optimizations will be conducted transparently to improve the end-to-end video query performance. 
+Under the hood, VQPy will automatically select an object detection model that outputs the specified `cls_name`, and decide the workflow to generate the object properties. Multiple video optimizations will be conducted transparently to improve the end-to-end video query performance.
 
-### Customization
+#### Detector and tracker
 
-For more details on customization, please refer to the document [here](https://github.com/uclasystem/VQPy/blob/main/docs/frontend.md#customization).
+Under normal conditions, users do not need to provide a detector implementation. However, if the user have a detector, they can inherit `vqpy.detector.DetectorBase` and implement the required interfaces.
+
+```python
+class DetectorBase(object):
+    """The base class of all predictors"""
+    cls_names = None        # the class names of the classification
+    output_fields = []      # the list of data fields the predictor can provide
+
+    def __init__(self, model_path: str) -> None:
+        self.model_path = model_path
+
+    def inference(self, img: np.ndarray) -> List[Dict]:
+        """Get the detected objects from the image
+        img (np.ndarray): the inferenced images
+        returns: list of objects, expressed in dictionaries
+        """
+        raise NotImplementedError
+```
+
+There are two level of trackers in VQPy. The surface level tracker stores the `VObj` instances by classes, and this is where `VObj` instances are created. For the default surface level tracker, different classes will be tracked separately using the ground level tracker. This level of tracker should generate the `track_id` field based on the provided detection result in this frame. Clearly, these trackers should be able to memorize historic detection results. To implement a method generating `track_id`, one should inherit `GroundTrackerBase` and register it to our backend. The interface of the ground level tracker is show below:
```

Comment:
Shall we separate the explanation for our design logic and how users should customize a tracker